morf-postgresql/src/main/java/org/alfasoftware/morf/jdbc/postgresql/PostgreSQLDialect.java

```diff
@@ -478,6 +485,43 @@ protected String getSqlFrom(SelectFirstStatement stmt) {
   }
 
 
+  @Override
+  protected String selectStatementPreFieldDirectives(SelectStatement selectStatement) {
+    StringBuilder builder = new StringBuilder();
+
+    for (Hint hint : selectStatement.getHints()) {
+      if (hint instanceof OptimiseForRowCount) {
+        // not available in pg_hint_plan
+      }
+      if (hint instanceof UseIndex) {
+        UseIndex useIndex = (UseIndex)hint;
+        builder.append(" IndexScan(")
+          .append(StringUtils.isEmpty(useIndex.getTable().getAlias()) ? useIndex.getTable().getName() : useIndex.getTable().getAlias())
+          .append(" ")
+          .append(useIndex.getIndexName().toLowerCase())
+          .append(")");
+      }
+      if (hint instanceof UseImplicitJoinOrder) {
+        // not available in pg_hint_plan
+        // actually, there is Leading hint, which we could abuse
+      }
+      if (hint instanceof ParallelQueryHint) {
+        // not available in pg_hint_plan
+      }
+      if (hint instanceof PostgreSQLCustomHint) {
+        builder.append(" ")
+        .append(((PostgreSQLCustomHint)hint).getCustomHint());
+      }
+    }
+
+    if (builder.length() == 0) {
+      return super.selectStatementPreFieldDirectives(selectStatement);
+    }
+
+    return "/*+" + builder.append(" */ ").toString();

```

Comment:
I assume. as a comment, that this will be ignored if pg_hint_plan is not being used.

Reply:
Yup. Beauty of hints via comments.